yjit/src/asm/mod.rs

```diff
@@ -121,10 +153,6 @@ impl CodeBlock {
         self.write_pos
     }
 
-    pub fn get_mem(&mut self) -> &mut VirtualMem {
-        &mut self.mem_block
-    }
-
     // Set the current write position
     pub fn set_pos(&mut self, pos: usize) {
         // No bounds check here since we can be out of bounds
```

Comment:
I was thinking we might want to eliminate the `set_pos` and `get_pos` methods that work with a `usize` index given that we might end up with code that's no longer contiguous in memory and this won't make sense anymore. What do you think?

Reply:
My assumption was that `set_pos` is used only for rewriting existing code and then called again to recover the original position. If that's true, a page_fault shouldn't happen when it's used. I think we need to add assertion or use interface that enforces it to guarantee that.

Reply:
Ok so you're assuming that get_pos get you the position within a page. I think that makes sense.